test/descriptor.dart

```diff
@@ -33,7 +33,7 @@ TarFileDescriptor tar(String name, [List<Descriptor>? contents]) =>
 
 /// Describes a package that passes all validation.
 DirectoryDescriptor get validPackage => dir(appPath, [
-      libPubspec('test_pkg', '1.0.0', sdk: '>=1.8.0 <=2.0.0'),
+      libPubspec('test_pkg', '1.0.0', sdk: '>=0.1.2 <=0.2.0'),
```

Comment:
We have a lot of tests testing publishing a "valid" package. The default sdk when running a test is v0.1.2+3, that went fine when `pub lish` did no resolution, but now it would fail. So Either we had to update each publish-test with a custom sdk version or to make the valid package compatible with the test version.

Will put a line in the pr description.